Feature/docker

abossenbroek opened the pull request:
This enables deployment of play on docker as well as integration with heroku. It also includes a `README.md` that explains how to build, push and publish a new version of the application. A `docker-compose.yml` is also included to allow to start a web server and db server.

### Preparation
To test this release first you will need docker,
 - [Install Docker](https://docs.docker.com/docker-for-mac/install/)

### Generate new release
To generate a new release of the application that can be run locally as well as on Heroku you can use sbt,
 - run `sbt docker:publishLocal`

### Test locally,
 - `docker run -p 9000:9000 -e APPLICATION_SECRET="changeme" --rm -ti framed-server:1.0-SNAPSHOT`
 - and point your browser to localhost:9000 to see the landing page

### Test on heroku,
 - `heroku container:login`
 - `docker tag framed-server:1.0-SNAPSHOT registry.heroku.com/framed/web`
 - `docker push registry.heroku.com/framed/web`
 - `heroku container:release web`
 - start on your terminal `heroku logs --tail` and point your browser to [the heroku page](https://framed.herokuapp.com) you should see something along these lines `app[web.1]: [info] play.api.Play - Application started (Prod)`

abossenbroek committed:
Removes obsolete filed

abossenbroek committed:
Moves most settings to secure.conf

Many configuration settings where in application.conf, these have now been moved to secure.conf since they were mostly concerned with security. This paves the way for easier unit testing

abossenbroek committed:
Bump SBT version to 1.2.7

abossenbroek committed:
Adds Docker support to SBT and docker compose

abossenbroek committed:
Adds README that explains how to build and start with docker (compose)

abossenbroek committed:
Adds short description of heroku deployment steps

abossenbroek committed:
Moves docker container settings to server settings

Also fixes library version for scalajs

abossenbroek committed:
Fixes explanation of Docker publishing

abossenbroek committed:
Include secure hosting to settings of heroku.conf

abossenbroek committed:
Changes SBT docker deployment to work with heroku

abossenbroek committed:
Adds documentation on releasing on heroku

manonja commented:
Great work on the deployment! 

I had to do small changes to test it. Maybe good if add them on the test locally section: 

- specify the type of port: `-e PORT=9000` we get now: `docker run -p 9000:9000 -e APPLICATION_SECRET="changeme" -e PORT=9000 --rm -ti framed-server:1.0-SNAPSHOT`

- in order to get the `APPLICATION_SECRET` code, type `sbt playGenerateSecret`. I got a number that I replaced: `docker run -p 9000:9000 -e APPLICATION_SECRET='to-be-filled-in-with-new-code' -e PORT=9000 --rm -ti framed-server:1.0-SNAPSHOT`

And it works well 👍 👯

abossenbroek committed:
Adds how to generate a --env-file

abossenbroek committed:
Adds .env file to docker-compose

abossenbroek committed:
Adds .env to gitignore

abossenbroek committed:
Adds how to start using an .env file

abossenbroek committed:
Removes ugly empty lines

abossenbroek commented:
Super feedback 😎! I found an easier way to work with environment variables. Instead of passing them on the command line and having to save the APPLICATION_SECRET somewhere for later use I now leverage an `.env` file.

A `.env` file allows storing environment variables that are only known to someone's local machine. I added the file to `.gitignore` to ensure that it is not added to the repository. 

I updated the `README.md` and the `docker-compose.yml` files to support the `.env` file.

You can now start docker locally using `docker run -p 9000:9000 --env-file .env --rm -ti framed-server:1.0-SNAPSHOT` or just `docker-compose up`
